windows-gaming-driver/src/config.rs

```diff
@@ -82,19 +186,39 @@ impl From<DeviceId> for (u16, u16) {
 
 impl Config {
     pub fn save<P: AsRef<Path>>(&self, path: P) {
-        let contents = toml::to_string(self).unwrap();
+        let contents = serde_yaml::to_string(self).unwrap();
         let mut file = OpenOptions::new().create(true).write(true)
             .truncate(true).open(path).expect("Failed to open config file");
-        write!(file, "{}", contents).expect("Failed to write config file");
+        writeln!(file, "{}", contents).expect("Failed to write config file");
     }
 
-    pub fn load<P: AsRef<Path>>(path: P) -> Config {
+    pub fn load<P: AsRef<Path>>(path: P) -> Option<Config> {
```

Comment:
If implementing config migration, we should do so properly. We will definitely need automatic config migration at some point, so why not do it now? :)

I'd suggest adding a "version" property to the config file. We should first read the version property if possiblea as struct having only that one property, or parse the config file as `Value` and manually extract the "version" information. We should create a new mod containing one function which reads the version and calls all proper upgrade functions, which are in (different) mods: `config/mod.rs` contains all current config stuff. Either `config/version.rs` or `config/version/mod.rs` contains the main upgrade function and either `config/version/version{1,2,…}.rs` or `config/version{1,2,…}.rs` contain the functions to upgrade from the previous to that version.